plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +505,201 @@ private function checkTroll(int $userID, array &$args) {
             $args['IsValid'] = false;
         }
     }
+
+    /**
+     * Add a configuration popup to the Troll Management plugin.
+     *
+     * @param Controller $sender
+     */
+    public function settingsController_trollManagement_create($sender) {
+        // Prevent non-admins from accessing this page
+        $sender->permission('Garden.Settings.Manage');
+
+        $sender->title(sprintf(t('%s settings'), t('Troll Management')));
+        $sender->setData('PluginDescription', $this->getPluginKey('Description'));
+
+        $sender->Form = new Gdn_Form();
+        $validation = new Gdn_Validation();
+        $configurationModel = new Gdn_ConfigurationModel($validation);
+
+        $configurationModel->setField([
+            'TrollManagement.PerFingerPrint.Enabled' => c('TrollManagement.PerFingerPrint.Enabled', false),
```

Comment:
This is a little nit-picky, but let's keep the casing of "fingerprint" consistent (e.g. `Fingerprint` versus `FingerPrint`).